### ISSUE TITLE
Training model using DataParallel and 2 GPU crashes system or causes gpu getting lost

### ISSUE THREAD

**Issue by saschalippert** [pytorch/pytorch#30998]
## 🐛 Bug

Training the model on two gpus using DataParallel results either in a crash (black screen) or GPU being listed in nvidia-smi as "lost". In the second case the python process will not terminate.

For testing I reduced the batch size to 10 and suddenly the training worked fine. (used batch size 32 before) Using batch size 32 the training crashed my system 100% of the time. To provoke a crash I increased the batch size to 64 and the training still works fine...

If the model is trained on gpu0 or gpu1 individually, the training works fine. Batch size has no impact in that case.

simpleP2P and p2pBandwidthLatencyTest pass fine (logs below)

When running a burn in test on windows no errors are encountered, so I assume the PSU (850W) is also not faulty.

## To Reproduce

**Relevant code:**

device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
model = nn.DataParallel(model).to(device)

images = images.to(device)
latents = latents.to(device)

outputs = model(images)

loss = criterion(outputs, latents)

**Logs:**

./simpleP2P 
[./simpleP2P] - Starting...
Checking for multiple GPUs...
CUDA-capable device count: 2

Checking GPU(s) for support of peer to peer memory access...
> Peer access from GeForce GTX 1080 Ti (GPU0) -> GeForce GTX 1080 Ti (GPU1) : Yes
> Peer access from GeForce GTX 1080 Ti (GPU1) -> GeForce GTX 1080 Ti (GPU0) : Yes
Enabling peer access between GPU0 and GPU1...
Allocating buffers (64MB on GPU0, GPU1 and CPU Host)...
Creating event handles...
cudaMemcpyPeer / cudaMemcpy between GPU0 and GPU1: 5.94GB/s
Preparing host buffer and memcpy to GPU0...
Run kernel on GPU1, taking source data from GPU0 and writing to GPU1...
Run kernel on GPU0, taking source data from GPU1 and writing to GPU0...
Copy data back to host from GPU0 and verify results...
Disabling peer access...
Shutting down...
Test passed

./p2pBandwidthLatencyTest 
[P2P (Peer-to-Peer) GPU Bandwidth Latency Test]
Device: 0, GeForce GTX 1080 Ti, pciBusID: 1, pciDeviceID: 0, pciDomainID:0
Device: 1, GeForce GTX 1080 Ti, pciBusID: 2, pciDeviceID: 0, pciDomainID:0
Device=0 CAN Access Peer Device=1
Device=1 CAN Access Peer Device=0

***NOTE: In case a device doesn't have P2P access to other one, it falls back to normal memcopy procedure.
So you can see lesser Bandwidth (GB/s) and unstable Latency (us) in those cases.

P2P Connectivity Matrix
     D\D     0     1
     0	     1     1
     1	     1     1
Unidirectional P2P=Disabled Bandwidth Matrix (GB/s)
   D\D     0      1 
     0 349.40   6.02 
     1   6.05 355.11 
Unidirectional P2P=Enabled Bandwidth (P2P Writes) Matrix (GB/s)
   D\D     0      1 
     0 352.55   6.37 
     1   6.39 355.11 
Bidirectional P2P=Disabled Bandwidth Matrix (GB/s)
   D\D     0      1 
     0 352.48   8.81 
     1   8.86 357.20 
Bidirectional P2P=Enabled Bandwidth Matrix (GB/s)
   D\D     0      1 
     0 350.97  12.71 
     1  12.70 354.58 
P2P=Disabled Latency Matrix (us)
   GPU     0      1 
     0   1.33  17.39 
     1  17.31   1.36 

   CPU     0      1 
     0   2.16   4.40 
     1   4.69   2.09 
P2P=Enabled Latency (P2P Writes) Matrix (us)
   GPU     0      1 
     0   1.34   1.03 
     1   1.01   1.36 

   CPU     0      1 
     0   2.13   1.40 
     1   1.42   2.11 

NOTE: The CUDA Samples are not meant for performance measurements. Results may vary when GPU Boost is enabled.

## Expected behavior

DataParallel training does not crash system or gpu

## Environment

Kernel: 4.15.0-72-generic

PyTorch version: 1.3.1
Is debug build: No
CUDA used to build PyTorch: 10.1.243

OS: Linux Mint 19.2 Tina
GCC version: (Ubuntu 7.4.0-1ubuntu1~18.04.1) 7.4.0
CMake version: Could not collect

Python version: 3.6
Is CUDA available: Yes
CUDA runtime version: Could not collect
GPU models and configuration: 
GPU 0: GeForce GTX 1080 Ti
GPU 1: GeForce GTX 1080 Ti

Nvidia driver version: 435.21
cuDNN version: Could not collect

Versions of relevant libraries:
[pip3] numpy==1.17.4
[pip3] torch==1.3.1
[pip3] torchvision==0.4.2
[conda] Could not collect

**Comment by saschalippert** [comment]
smi while training with batch size 13

Mon Dec  9 22:13:44 2019       
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 435.21       Driver Version: 435.21       CUDA Version: 10.1     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|===============================+======================+======================|
|   0  GeForce GTX 108...  Off  | 00000000:01:00.0  On |                  N/A |
| 46%   80C    P2   276W / 250W |   4367MiB / 11175MiB |    100%      Default |
+-------------------------------+----------------------+----------------------+
|   1  GeForce GTX 108...  Off  | 00000000:02:00.0 Off |                  N/A |
| 36%   64C    P2   287W / 250W |   2041MiB / 11178MiB |     78%      Default |
+-------------------------------+----------------------+----------------------+

**Comment by saschalippert** [comment]
before decreasing and increasing the batch size again step by step, the error was occurring 100% at the time when training straight after booting the system. i even had the same issue running in arch linux. first though i was distro/kernel related. so i installed a fresh linux mint.

now even after a reboot i was not able to reproduce the error with batch size 128 under mint. i am not sure how to make sense of this. maybe its an issue related to low gpu/psu temperature?

some more smi logs:

Mon Dec  9 22:34:24 2019       
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 435.21       Driver Version: 435.21       CUDA Version: 10.1     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|===============================+======================+======================|
|   0  GeForce GTX 108...  Off  | 00000000:01:00.0  On |                  N/A |
| 49%   84C    P2   239W / 250W |  10909MiB / 11175MiB |     99%      Default |
+-------------------------------+----------------------+----------------------+
|   1  GeForce GTX 108...  Off  | 00000000:02:00.0 Off |                  N/A |
| 40%   71C    P2   199W / 250W |  10049MiB / 11178MiB |     99%      Default |
+-------------------------------+----------------------+----------------------+

**Comment by saschalippert** [comment]
since its very likely psu related (maybe lower efficiency at low temp) i am closing the issue

**Comment by devendraswamy** [comment]
please suggest any solution for this problem i am also facing the issue like this ??
i am given batch size = 2 and no of workers are 2 only and i am try to train the model in three gpus (each one had 24GB memory). with in few minutes my system will restarted , so please give me any suggestions and tell me is happen due to code or any hardware issue ??

Thanking you !

**Comment by saschalippert** [comment]
i upgraded my psu from 850w to 1000w. this solved the issue.

**Comment by devendraswamy** [comment]
Thank you for your response and is there any another solutions , please let
me know. Thanking you.

**Comment by saschalippert** [comment]
you could try to limit the power consumption, disable boosting or underclock your gpu.

http://developer.download.nvidia.com/compute/DCGM/docs/nvidia-smi-367.38.pdf

**Comment by devendraswamy** [comment]
Thank so much sir , I will try and then let me inform you !
